main.py

```diff
@@ -60,13 +60,18 @@ def signup():
 
     if form.validate_on_submit():
         hashedPassword = generate_password_hash(form.password.data, method='sha256')
-        newUser = User(name=form.name.data, email= form.email.data, password=hashedPassword)
 
-        db.session.add(newUser)
-        db.session.commit()
+        isUserExisting = User.query.filter_by(email=form.email.data).first()
 
-        return redirect(url_for('login'))
+        if not isUserExisting:
+            newUser = User(name=form.name.data, email= form.email.data, password=hashedPassword)
 
+            db.session.add(newUser)
+            db.session.commit()
+
+            return redirect(url_for('login'))
+            
+        flash("A user already exists with that email!")
```

Comment:
The logic is pretty simple here. If the user does not exist, create the new user and commit to the database. Otherwise, flash an error message and redirect back to the signup page.